berkdb/dbreg/dbreg_rec.c

```diff
@@ -100,13 +100,15 @@ __dbreg_register_recover(dbenv, dbtp, lsnp, op, info)
 		int i;
 		ft = &dbenv->fileid_track;
 		if (argp->fileid >= ft->numids) {
-			__os_realloc(dbenv, (argp->fileid+1) * 
```

Comment:
It would make this change easier to read if you added a local variable with the new allocation size right inside this "if" statement, e.g.: "int newNumids = argp->fileid * 2 + 1;" and then make use of that instead of re-calculating it later.